### DISCUSSION TITLE
Database error

### DISCUSSION THREAD

**Discussion by esambu** [apache/superset#13169] [Q&A / Help]
How can correct the error below am experiencing while to trying to preview a table or make an sql query

Database error
Instance <Query at 0x21a5cdd15e0> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)

**Comment by esambu** [comment]
Am using windows with superset version 1.0.1 with the above error

**Reply by esambu** [reply]
I just got the solution for my error on **Error running SQL Lab
#12824**

**Reply by pratik-sankey** [reply]
hey @esambu , I am facing same error. can you share me solution you got?

**Comment by esambu** [comment]
The problem seems to be caused by SIGALRM. Here is a workaround which has
worked for me.
In Lib\site-packages\superset\utils\core.py replace:

  def __enter__(self):
      try:
          signal.signal(signal.SIGALRM, self.handle_timeout)
          signal.alarm(self.seconds)
      except ValueError as e:
          logger.warning("timeout can't be used in the current context")
          logger.exception(e)

  def __exit__(self, type, value, traceback):
      try:
          signal.alarm(0)
      except ValueError as e:
          logger.warning("timeout can't be used in the current context")
          logger.exception(e)


with

    def __enter__(self):
        try:
            #signal.signal(signal.SIGALRM, self.handle_timeout)
            #signal.alarm(self.seconds)
            pass
        except ValueError as e:
            logger.warning("timeout can't be used in the current context")
            logger.exception(e)

    def __exit__(self, type, value, traceback):
        try:
            #signal.alarm(0)
            pass
        except ValueError as e:
            logger.warning("timeout can't be used in the current context")
            logger.exception(e)
